web/src/message_fetch.ts

```diff
@@ -794,3 +807,43 @@ export function initialize(finished_initial_fetch: () => void): void {
         cont: load_more,
     });
 }
+
+export function get_narrow_summary(channel_id: number, topic_name: string): void {
+    const filter = new Filter([
+        {operator: "channel", operand: `${channel_id}`},
+        {operator: "topic", operand: topic_name},
+    ]);
+    const data = {narrow: get_narrow_for_message_fetch(filter)};
+    const channel_name = stream_data.get_stream_name_from_id(channel_id);
+    dialog_widget.launch({
+        html_heading: `Summary of #${channel_name} > ${topic_name}:`,
```

Comment:
This code is an XSS vulnerability. I know you were prototyping, but my personal policy is to never write code like this even in prototypes, because the goal is for those to become actual production code, and it's easy to forget to check this sort of thing. 

Using our translation system protects against that, as does never passing user input directly into something with `html` in its name.